### PR TITLE
LPS-87488 rename module ext deployed file

### PR DESCRIPTION
**Pull request by Camork**

### CODE REVIEW

**Comment by gamerson** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/sdk/gradle-plugins/src/main/java/com/liferay/gradle/plugins/LiferayOSGiExtPlugin.java [hunk 1]
```diff
@@ -222,11 +223,34 @@ private void _configureProject(
...
+			jar,
+			new Closure<Void>(project) {
+
+				@SuppressWarnings("unused")
+				public void doCall(CopySpec copySpec) {
+					copySpec.rename(
+						new Closure<String>(project) {
+
+							public String doCall(String fileName) {
+								Closure<String> deployedFileNameClosure =
```
Comment (gamerson) [inline modules/sdk/gradle-plugins/src/main/java/com/liferay/gradle/plugins/LiferayOSGiExtPlugin.java]:
```suggestion
								Closure<String> closure =
```

**Comment by gamerson** [comment]
hey @Camork  this seems incomplete.

We need to do one of the following:
1. in the _configureLiferay we should do something like
```
liferayExtension.setDeployedFilenameClosure(...)
```
To automatically set the correct deployedFilename as "originalModuleName".jar
2. if we don't do 1, then we should modify the Modules-Ext project template to add this in the build.gradle
```
liferay {
        deployedFileNameClosure {
                "com.liferay.login.web.jar"
        }
}
```
templatized
```
liferay {
        deployedFileNameClosure {
                "${originalModuleName}.jar"
        }
}
```

**Comment by Camork** [comment]
Hey @gamerson, i can't get the point, the default `deployedFileNameClosure` in [LiferayExtension](https://github.com/liferay/liferay-portal/blob/master/modules/sdk/gradle-plugins/src/main/java/com/liferay/gradle/plugins/extensions/LiferayExtension.java#L89) is exactly what we want, why we need to set it again?

**Comment by gamerson** [comment]
hey @Camork  I pulled in your code, applied the plugin, and I still see the old filename.  I don't see how your code correctly renames the file.  In my test the file that is added to /osgi/marketplace/override is still the same 'com.liferay.login.web-3.0.2.jar' or whatever, instead of 'com.liferay.login.web.jar'.  So I don't see how this change fixes anything.

Now if I do this
```
liferay {
    deployedFileNameClosure {
                "com.liferay.login.web.jar"
        }
}
```

Then it works.  However, if this is indeed required we need to make this option generated in our template.  

Or we need to make this happen without the need of the deployedFileNameClosure at all (we just automatically adding the closure).  

Am I missing something?

**Comment by gamerson** [comment]
> Hey @gamerson, i can't get the point, the default `deployedFileNameClosure` in [LiferayExtension](https://github.com/liferay/liferay-portal/blob/master/modules/sdk/gradle-plugins/src/main/java/com/liferay/gradle/plugins/extensions/LiferayExtension.java#L89) is exactly what we want, why we need to set it again?

What is the default filename closure again?  I don't see that the default is working for me.

**Comment by Camork** [comment]
That's weird we get different result.
 maybe you miss set 
`classpath group: "com.liferay", name: "com.liferay.gradle.plugins", version: "3.13.27"` or 
`maven { url "file:///D:/git/liferay-portal/.m2-tmp/" }` ? 

By the way, i added the test to verify this, and the gradle cache was tested locally in my side.

In the code side [LiferayExtension](https://github.com/liferay/liferay-portal/blob/master/modules/sdk/gradle-plugins/src/main/java/com/liferay/gradle/plugins/extensions/LiferayExtension.java#L89)  Line 89, it had been set for LiferayExtension, this commit only make this closure work (that's how other module work by [LiferayOSGiPlugin](https://github.com/liferay/liferay-portal/blob/8b075f1c9f2629d879ba42580ec4e4058a1e25a8/modules/sdk/gradle-plugins/src/main/java/com/liferay/gradle/plugins/LiferayOSGiPlugin.java#L272)).

**Comment by gamerson** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by gamerson** [comment]
Pull request submitted to https://github.com/petershin/liferay-portal/pull/328. [See changes here.](https://github.com/gamerson/liferay-portal/compare/e6f57c80843b~1...e6f57c80843b) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Pull request closed without merging**